dotnet/src/SemanticKernel.UnitTests/Utilities/SseJsonParserTests.cs

```diff
@@ -0,0 +1,211 @@
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Text;
+using Xunit;
+
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes
+
+namespace SemanticKernel.UnitTests.Utilities;
+
+public sealed class SseJsonParserTests
+{
+    public const string SampleSseData1 =
+        """
+        event: message_start
+        data: {"type": "message_start", "message": {"id": "msg_1nZdL29xx5MUA1yADyHTEsnR8uuvGzszyY", "type": "message", "role": "assistant", "content": [], "model": "claude-3-opus-20240229", "stop_reason": null, "stop_sequence": null, "usage": {"input_tokens": 25, "output_tokens": 1}}}
+
+        event: content_block_start
+        data: {"type": "content_block_start", "index": 0, "content_block": {"type": "text", "text": ""}}
+
+        event: ping
+        data: {"type": "ping"}
+
+        event: content_block_delta
+        data: {"type": "content_block_delta", "index": 0, "delta": {"type": "text_delta", "text": "Hello"}}
+
+        event: content_block_delta
+        data: {"type": "content_block_delta", "index": 0, "delta": {"type": "text_delta", "text": "!"}}
+
+        event: content_block_stop
+        data: {"type": "content_block_stop", "index": 0}
+
+        event: message_delta
+        data: {"type": "message_delta", "delta": {"stop_reason": "end_turn", "stop_sequence":null, "usage":{"output_tokens": 15}}}
+
+        event: message_stop
+        data: {"type": "message_stop"}
+
+        """;
+
+    public const string SampleSseData2 =
+        """
+        event: userconnect
+        data: {"username": "bobby", "time": "02:33:48"}
+
+        event: usermessage
+        data: {"username": "bobby", "time": "02:34:11", "text": "Hi everyone."}
+
+        event: userdisconnect
+        data: {"username": "bobby", "time": "02:34:23"}
+
+        event: usermessage
+        data: {"username": "sean", "time": "02:34:36", "text": "Bye, bobby."}
+        """;
+
+    public const string SampleSseData3 =
+        """
+        event: userconnect
+        data: {"username": "bobby", "time": "02:33:48"}
+
+        data: Here's a system message of some kind that will get used
+        data: to accomplish some task.
+
+        event: usermessage
+        data: {"username": "bobby", "time": "02:34:11", "text": "Hi everyone."}
+        """;
+
+    public const string SampleSseData4 =
+        """
+        event: userconnect
+        data: {"username": "bobby", "time": "02:33:48"}
+
+        data: none
+
+        event: usermessage
+        data: {"username": "bobby", "time": "02:34:11", "text": "Hi everyone."}
+
+        event: userdisconnect
+        data: {"username": "bobby", "time": "02:34:23"}
+        data:
+        data
+        id: 3
+
+        data: [DONE]
+
+        event: usermessage
+        data: {"username": "sean", "time": "02:34:36", "text": "Bye, bobby."}
+
+        """;
```

Comment:
Could we have some sample data the looks like the response from an LLM?

Also consider moving sample data into .txt files in a seperate folder.

Reply:
SampleSseData1 is sample from Claude response.